source/slang/slang-emit.cpp

```diff
@@ -604,19 +608,23 @@ Result linkAndOptimizeIR(
     return SLANG_OK;
 }
 
+// TODO(DG): This probably needs to be generalized to a list
 SlangResult emitEntryPointSourceFromIR(
     BackEndCompileRequest*  compileRequest,
-    Int                     entryPointIndex,
+    List<Int>               entryPointIndices,
     CodeGenTarget           target,
     TargetRequest*          targetRequest,
     SourceResult&       outSource)
 {
+    // Temporary assertion for checkpoint
+    SLANG_ASSERT(entryPointIndices.getCount() == 1);
     outSource.reset();
 
     auto sink = compileRequest->getSink();
     auto program = compileRequest->getProgram();
 
-    auto entryPoint = program->getEntryPoint(entryPointIndex);
+    // TODO(DG): Update from assertion
+    auto entryPoint = program->getEntryPoint(entryPointIndices[0]);
```

Comment:
Style-wise, I think it works a lot better to do:

```
SLANG_ASSERT(entryPointIndices.getCount() == 1);
auto entryPointIndex = entryPointIndices[0];
```

at the top of functions like this, and thereby avoid having to change any of the existing references to `entryPointIndex` downstream.